tests/kernel/semaphore/semaphore/src/main.c

```diff
@@ -1025,39 +906,40 @@ void test_sem_measure_timeout_from_thread(void)
 
 
 	/* first sync the 2 threads */
-	k_sem_take(&simple_sem, K_FOREVER);
+	expect_k_sem_take_nomsg(&simple_sem, K_FOREVER, 0);
 
 	/* with timeout of 1 sec */
 	start_ticks = k_uptime_get();
 
-	ret_value = k_sem_take(&multiple_thread_sem, K_SECONDS(1));
+	expect_k_sem_take_nomsg(&multiple_thread_sem, K_SECONDS(1), 0);
 
 	end_ticks = k_uptime_get();
 
-	zassert_true(ret_value == 0, "k_sem_take failed");
+	diff_ticks = end_ticks - start_ticks;
 
-	zassert_true((end_ticks - start_ticks <= SEC2MS(1)),
-		     "time missmatch - expected less than%d ,got %d",
-		     SEC2MS(1), end_ticks - start_ticks);
+	zassert_true((diff_ticks < SEC2MS(1)),
+		     "k_sem_take took too long: %d >= %d",
+		     diff_ticks, SEC2MS(1));
 
 }
 
 void sem_multiple_take_and_timeouts_helper(void *p1, void *p2, void *p3)
 {
 	int timeout = POINTER_TO_INT(p1);
-	uint32_t start_ticks, end_ticks;
+	int64_t start_ticks, end_ticks, diff_ticks;
```

Comment:
Pedantry: there is a k_ticks_t defined now for this purpose.  The kernel's tick representation (and thus the longest expressible timeout) can be configured to be 32 or 64 bit.

Reply:
> Pedantry: there is a k_ticks_t defined now for this purpose. The kernel's tick representation (and thus the longest expressible timeout) can be configured to be 32 or 64 bit.

Hm. What's the `k_uptime_get` equivalent that returns k_ticks_t? I can't seem to find it.

Reply:
k_uptime_ticks(), really that's the underlying API (likewise xxx_remaining_ticks() for the various user-visible timeout sources).  The ms API is legacy, but obviously useful enough that it isn't going away.  Time was, we had a somewhat confused mix of ticks and ms in the API reaching way down into the kernel, which was a big problem for platforms like nRF5 whose 32768 Hz timer doesn't evenly divide milliseconds.